packages/core/elasticsearch/core-elasticsearch-server-internal/src/elasticsearch_config.ts

```diff
@@ -154,7 +154,8 @@ export const configSchema = schema.object({
           return '"ignoreVersionMismatch" can only be set to true in development mode';
         }
       },
-      defaultValue: false,
+      // When running in serverless mode, default to `true`
+      defaultValue: schema.contextRef('serverless'),
```

Comment:
I thought about nesting another conditional but found this solution a bit cleaner. Happy to iterate if you think it's best to have the explicit conditional.

Reply:
This is fine as long as we're absolutely certain there aren't any other valid use cases where the version mismatch doesn't matter. If we're not certain, then a nested conditional is safer.

Reply:
Yeah, fine for now ihmo, we can change later if needed.